
syntax = "proto3";

message NullMessage {}

message Step {
    fixed64 n = 1;
}

message NdArrays {
    message NdArray {
        string dtype = 1;
        repeated int32 shape = 2;
        bytes data = 3;
    }
    repeated NdArray arrays = 1;
}

service Master {
    rpc IncrementGlobalT(NullMessage) returns (Step) {}
    rpc ApplyGradients(NdArrays) returns (NullMessage) {}
    rpc GetValues(NullMessage) returns (NdArrays) {}
}
